torch/_inductor/codecache.py

```diff
@@ -379,6 +388,163 @@ def write_atomic(path: str, content: Union[str, bytes]) -> None:
     tmp_path.rename(path)
 
 
+def _reduce_fake_tensor(t):
+    """
+    See FxGraphCachePickler. Custom reducer to enable pickling FakeTensor.
+    """
+    return (TensorMetadataHolder, (_extract_tensor_metadata(t), t.device))
+
+
+@dataclasses.dataclass
```

Comment:
I got this idea from here: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/debug.py#L475-L505